drivers/instr-amplif/ada4250/ada4250.c

```diff
@@ -0,0 +1,304 @@
+#include "error.h"
+
+/******************************************************************************/
+/************************** Functions Implementation **************************/
+/******************************************************************************/
+/**
+ * @brief Writes data to ada4250 over SPI.
+ * @param dev - The device structure.
+ * @param reg_addr - The register address.
+ * @param data - Data value to write.
+ * @return Returns SUCCESS in case of success or negative error code otherwise.
+ */
+
+int32_t ada4250_write(struct ada4250_dev *dev, uint8_t reg_addr,
+		      uint8_t data)
+{
+	uint8_t buff[ADA4250_BUFF_SIZE_BYTES];
+
+	buff[0] = ADA4250_SPI_WRITE_CMD | reg_addr;
+	buff[1] = data;
+
+	return spi_write_and_read(dev->spi_desc, buff, ADA4250_BUFF_SIZE_BYTES);
+}
+
+/**
+ * @brief Reads data from ada4250 over SPI.
+ * @param dev - The device structure.
+ * @param reg_addr - The register address.
+ * @param data - Data read from the device.
+ * @return Returns SUCCESS in case of success or negative error code otherwise.
+ */
+
+int32_t ada4250_read(struct ada4250_dev *dev, uint8_t reg_addr,
+		     uint8_t *data)
+{
+	int32_t ret;
+	uint8_t buff[ADA4250_BUFF_SIZE_BYTES];
+
+	buff[0] = ADA4250_SPI_READ_CMD | reg_addr;
+	buff[1] = ADA4250_SPI_DUMMY_DATA;
+
+	ret = spi_write_and_read(dev->spi_desc, buff, ADA4250_BUFF_SIZE_BYTES);
+	if(ret != SUCCESS)
+		return ret;
+
+	*data = buff[1];
+
+	return ret;
+}
+
+/**
+ * @brief Update ADA4250 register.
+ * @param dev - The device structure.
+ * @param reg_addr - The register address.
+ * @param mask - Mask for specific register bits to be updated.
+ * @param data - Data read from the device.
+ * @return Returns SUCCESS in case of success or negative error code otherwise.
+ */
+int32_t ada4250_update(struct ada4250_dev *dev, uint8_t reg_addr,
+		       uint8_t mask, uint8_t data)
+{
+	uint8_t read_val;
+	int32_t ret;
+
+	ret = ada4250_read(dev, reg_addr, &read_val);
+	if (ret != SUCCESS)
+		return ret;
+
+	read_val &= ~mask;
+	read_val |= data;
+
+	ret = ada4250_write(dev, reg_addr, read_val);
+
+	return ret;
+}
+
+/**
+ * @brief Software reset.
+ * @param dev - The device structure.
+ * @return Returns SUCCESS in case of success or negative error code.
+ */
+int32_t ada4250_soft_reset(struct ada4250_dev *dev)
+{
+	int32_t ret;
+	uint16_t timeout = 0xFFFF;
+	uint8_t data;
+
+	ret = ada4250_update(dev, ADA4250_REG_RESET, ADA4250_RESET_MSK,
+			     ADA4250_RESET(ADA4250_RESET_ENABLE));
+
+	while(timeout--) {
+		ret = ada4250_read(dev, ADA4250_REG_RESET, &data);
+		if (ret != SUCCESS)
+			return ret;
+
+		if(!(data & ADA4250_RESET(ADA4250_RESET_ENABLE)))
+			return SUCCESS;
+	}
+
+	return FAILURE;
+}
+
+/**
+ * @brief Enable/Disable Reference Buffer.
+ * @param dev - The device structure.
+ * @param refbuf - REFBUF enable/disable.
+ * @return Returns SUCCESS in case of success or negative error code.
+ */
+int32_t ada4250_set_refbuf(struct ada4250_dev *dev, uint8_t refbuf)
```

Comment:
I guess "uint8_t refbuf" is the same thing as "bool refbuf_en" from ada4250_init_param  and ada4250_dev. If this is correct, we should use "bool" instead of "uint8_t" and name the function more intuitive (e.g., en_refbuf).

Reply:
Addressed with: https://github.com/analogdevicesinc/no-OS/pull/984